### PR TITLE
Extra Tests

### PR DESCRIPTION
**Pull request by kellen-martin**
Adds more tests using enzyme

### CODE REVIEW

File: extras/newton_enz.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// define variables
+double tol = 1;
+int n = 0;
+double xs[n_max];
+xs[0] = guess;
+
+printf("|  n  |  x  |  f(x)  |\n|  %i  |  %f  |  %e  |\n", n, xs[n], func(xs[n]));
+// Newton's 
+while (tol > target_tol && n < n_max)
+{
```
Comment (LeilaGhaffari) [inline extras/newton_enz.c:28]: Thanks, @kellen-martin , this is great. I don't really have a style convention in this repo but I would recommend sticking with one style in your commits (and also this is a good practice for reviewing PRs). Anyways, in your `for-loops` you open the curly brace right after the condition, but here, you open it on a new line. You decide which one you'd prefer.

**Review by LeilaGhaffari — CHANGES_REQUESTED** [review]

**Review by jedbrown — COMMENTED** [review]
Looking good. Can you add a `Makefile` to this directory to automate building? There is a simple one at `libceed-enzyme/Makefile` that can be copied over.

For formatting the C code, one option to ensure consistency is to grab [Ratel's `.clang-format`](https://gitlab.com/micromorph/ratel/-/blob/main/.clang-format), in which case vscode/clangd can auto-format.

**Comment by kellen-martin** [comment]
Thanks for the help @jedbrown & @LeilaGhaffari, I have added the Makefile and reformatted using .clangformat from Ratel.

**Comment by LeilaGhaffari** [comment]
Thanks, @kellen-martin for adding those files. Could you move `my-enzyme/extras/.clang-format` to `my-enzyme/` so it can impact other folders as well.

**Comment by LeilaGhaffari** [comment]
Oh, you can remove `my-enzyme/extras/.clang-format` now.

**Comment by LeilaGhaffari** [comment]
Great, thanks! Could you rebase your branch onto main before we merge it?
You can do it with `git pull --rebase origin main`.

**Comment by kellen-martin** [comment]
Ready for final review!

**Review by LeilaGhaffari — APPROVED** [review]
Thanks, @kellen-martin . It looks great! You can merge it now.

**Pull request merged**